sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TablesSharedKeyCredential.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.tables;
+
+import com.azure.storage.common.implementation.StorageImplUtils;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A Class which helps generate the shared key credentials for a given storage account to create a Http requests to
+ * access Azure Tables
+ */
+public class TablesSharedKeyCredential {
+    private static final String AUTHORIZATION_HEADER_FORMAT = "SharedKeyLite %s:%s";
+    private final String accountName;
+    private final String accountKey;
+
+    /**
+     * Constructor for TableSharedKeyCredential Class
+     *
+     * @param accountName name of the storage account
+     * @param accountKey key to the storage account
+     */
+    public TablesSharedKeyCredential(String accountName, String accountKey) {
+        Objects.requireNonNull(accountName, "'accountName' cannot be null.");
```

Comment:
This can be simplified to. Same with the line below.

```java
this.accountName = Objects.requireNonNull(accountName, "'accountName' cannot be null.");
```